Artesian/_GMEPublicOffers/QueryParameters/GMEPOfferQueryParameters.py

```diff
@@ -1,5 +1,54 @@
+
+
+from Artesian._GMEPublicOffers.Config.BaType import BaType
+from Artesian._GMEPublicOffers.Config.ExtractionRangeConfig import ExtractionRangeConfig
+from Artesian._GMEPublicOffers.Config.GenerationType import GenerationType
+from Artesian._GMEPublicOffers.Config.Market import Market
+from Artesian._GMEPublicOffers.Config.Purpose import Purpose
+from Artesian._GMEPublicOffers.Config.Scope import Scope
+from Artesian._GMEPublicOffers.Config.Status import Status
+from Artesian._GMEPublicOffers.Config.UnitType import UnitType
+from Artesian._GMEPublicOffers.Config.Zone import Zone
+
+
 class GMEPOfferQueryParameters: 
-    def __init__(self, scope, extractionRangeSelectionConfig, status, unitType, generationType,operator,unit,zone,market,purpose, page, pageSize,baType):
+    """ Class for the GME Public Offer Query Parameters.
+    
+        Attributes:
+            scope: sets scope to be queried
+            extractionRangeSelectionConfig: Sets the Extraction Range Selection Configuration to be queried.
+            status: sets the Status to be queried.
+            unitType: sets the unit types to be queried.
+            generationType: that sets the generation type to be queried.
+            operator: sets the operators to be queried.
+            unit: sets the units to be queried.
+            zone: sets the zones to be queried.
+            market: sets the Market to be queried.
+            purpose: sets the Purpose to be queried.
+            page: sets the Page to be queried.
+            pageSize: sets the Page size to be queried.
+            baType: sets the BATypes to be queried."""
+
+
+    def __init__(self, scope: Scope, extractionRangeSelectionConfig: ExtractionRangeConfig, status: Status, unitType: UnitType, generationType: GenerationType, operator: str, unit: str, zone: Zone , market: Market, purpose: Purpose, page: int, pageSize: int, baType: BaType)  -> None : 
```

Comment:
if possible try wrapping the params on multiple lines. not sure if pythons permit it, but I think so.